json_prolog/src/json_prolog/json_prolog.py

```diff
@@ -1,103 +1,125 @@
-
 import json
+from multiprocessing import Lock, TimeoutError
+
 import rospy
 from json_prolog_msgs import srv
 
-_simple_query_srv = None
-_next_solution_srv = None
-_finish_query_srv = None
+# We have to lock because calling json prolog twice at the same time can lead to a deadlock
```

Comment:
That's not true. Only if you call a json_prolog query **within** another one you will get a deadlock. That's because json_prolog is locking.

Reply:
Ok, but it doesn't change the fact that we need a lock right?

Reply:
The java implementation of json_prolog locks every query anyway. A second query has to wait until the first has finished. So locking in clients is not needed.

Reply:
oh I see and thats the reason for the deadlock, I'll remove the lock.

Reply:
Yes it is :( JPL documentation explicitly states that the calls must be locked.